src/server/JasmineGraphInstanceService.cpp

```diff
@@ -584,6 +587,271 @@ void *instanceservicesession(void *dummyPt) {
+
+            for (its = degreeDistributionCentral.begin(); its != degreeDistributionCentral.end();++its) {
+
+                bool centralNodeFound = false;
+                for (itcentral = degreeDistribution.begin(); itcentral != degreeDistribution.end();++itcentral) {
+
+                    if ((its->first) == (itcentral->first)) {
+                        degreeDistribution[its->first] = (its->second) + (itcentral->second);
+                        centralNodeFound = true;
+                    }
+                }
+                if (!centralNodeFound) {
+                    degreeDistribution.insert(std::make_pair(its->first, its->second));
+                }
+            }
+
+            // Invoke other workers to calculate their own our degree distributions
+            //todo  invoke other workers asynchronously
+            for (vector<string>::iterator workerIt=workerSockets.begin(); workerIt!=workerSockets.end(); ++workerIt) {
+                instance_logger.log("Worker pair " + *workerIt, "info");
+
+                std::vector <string> workerSocketPair;
+                boost::split(workerSocketPair, *workerIt, boost::is_any_of(":"));
+
+                if  (std::to_string(serverPort).compare(workerSocketPair[1]) == 0) {
+                    continue;
+                }
+
+                string host = workerSocketPair[0];
+                int port = stoi(workerSocketPair[1]);
+                int sockfd;
+                char data[300];
+                bool loop = false;
+                socklen_t len;
+                struct sockaddr_in serv_addr;
+                struct hostent *server;
+
+                sockfd = socket(AF_INET, SOCK_STREAM, 0);
+
+                if (sockfd < 0) {
+                    std::cout << "Cannot accept connection" << std::endl;
+                }
+
+                server = gethostbyname(host.c_str());
+                if (server == NULL) {
+                    std::cout << "ERROR, no host named " << server << std::endl;
+                }
+
+                bzero((char *) &serv_addr, sizeof(serv_addr));
+                serv_addr.sin_family = AF_INET;
+                bcopy((char *) server->h_addr,
+                      (char *) &serv_addr.sin_addr.s_addr,
+                      server->h_length);
+                serv_addr.sin_port = htons(port);
+                if (connect(sockfd, (struct sockaddr *) &serv_addr, sizeof(serv_addr)) < 0) {
+                    std::cout << "ERROR connecting" << std::endl;
+                    //TODO::exit
+                }
+
+                bzero(data, 301);
+                int result_wr = write(sockfd, JasmineGraphInstanceProtocol::SEND_IN_DEGREE_DISTRIBUTION_TO_AGGREGATOR.c_str(),
+                                      JasmineGraphInstanceProtocol::SEND_IN_DEGREE_DISTRIBUTION_TO_AGGREGATOR.size());
+
+                if(result_wr < 0) {
+                    instance_logger.log("Error writing to socket", "error");
```

Comment:
If there is an error I do not see a point of continuing further.

Reply:
added a returned statement

Reply:
Done